spec/factories/trainees.rb

```diff
@@ -1,17 +1,51 @@
 FactoryBot.define do
   factory :trainee do
+    sequence(:trainee_id) { |n| n.to_s }
+    first_names { Faker::Name.first_name }
+    last_name { Faker::Name.last_name }
+    gender { %w[Female Male].sample }
+    date_of_birth { Faker::Date.birthday(min_age: 18, max_age: 65) }
+    nationality { Faker::Nation.nationality }
+    ethnicity { Faker::Nation.nationality }
+    disability { %w[none something].sample }
+
+    a_level_1_subject { "Maths" }
+    a_level_1_grade { %w[A B C].sample }
+    a_level_2_subject { "Geography" }
+    a_level_2_grade { %w[A B C].sample }
+    a_level_3_subject { "Physics" }
+    a_level_3_grade { %w[A B C].sample }
+    degree_subject { "Software Engineering" }
+    degree_class { "2:1" }
+    degree_institution { "Durham University" }
+    degree_type { "BSc" }
+    ske { "SKE not required" }
+    previous_qts { false }
   end
 
   factory :trainee_for_form, class: Trainee do
```

Comment:
What's the diff between this `:trainee_for_form` vs `:trainee`?

Maybe a trait instead?

Reply:
it's do with the date. I've tidied up a little with a base/abstract class type thing. the form take 3 text field whereas the model takes an actual date object